src/ansys/simai/core/data/training_data.py

```diff
@@ -88,6 +89,11 @@ def get_subset(self, project: Identifiable["Project"]) -> Optional[str]:
         project_id = get_id_from_identifiable(project, default=self._client._current_project)
         return self._client._api.get_training_data_subset(project_id, self.id).get("subset")
 
+    def set_subset(self, project: Identifiable["Project"], subset: SubsetEnum) -> None:
```

Comment:
The method name seems confusing to me, maybe we should be more specific. If we are defining a type for an existing subset of data, I would suggest to rename the method to integrate the word **type** in it.

Reply:
This is confusing because we're in the `TrainingData` object but we're editing the `ModelTrainingDataAssociation`.
Maybe it could be named something like `set_project_training_subset` ?